js/common.js

```diff
@@ -0,0 +1,46 @@
+/* 에러 메시지 처리 */
+function wrongInput(input, span, validateFunction) {
```

Comment:
💊 제안
추상화를 하신 점이 좋습니다.
다만 함수명과 인자의 이름이 더 명확하면 좋을 것 같습니다.
해당 함수의 경우 에러메시지를 보여주고 안 보여주고 하는 역할을 하므로 
checkInputValid, toggleInputErrorMessage 와 같은 이름을 추천드려요.

인자의 경우 span 보다는 message가 보여질 태그임을 알 수 있는 이름으로 변경하시는 것을 추천드립니다~

```suggestion
function checkInputValid(input, messageTarget, validateFunction) {
```

Reply:
또한 validateFunction 을 인자로 받으셔서 실행하시는 것이 전부이니 아래처럼 하셔도 될 것 같아요~

```js
function checkInputValid(input, messageTarget, errorMessage) { 
    input.classList.toggle("warning", errorMessage);
    input.classList.toggle("done", !errorMessage);
    messageTarget.textContent = errorMessage;

    // TODO: span 태그를 무조건 추가하고 추후 지우는 방식이 아닌 방식을 고민해보세요~ 불필요한 DOM 조작은 성능에 좋지 않습니다.
    input.parentElement.appendChild(messageTarget);

    if (span.textContent === "") {
       input.parentElement.removeChild(messageTarget);
    }
}

checkInputValid(emailInput, wrongEmail, validateEmail(emailInput))
```